base/cvd/cuttlefish/host/libs/web/android_build_api.cpp

```diff
@@ -235,6 +239,14 @@ Result<std::string> BuildApi::BuildStatus(const DeviceBuild& build) {
   if (!api_key_.empty()) {
     url += "?key=" + http_client->UrlEscape(api_key_);
   }
+  if(!project_id_.empty()){
+    if (api_key_.empty()) {
+      url += "?$userProject=" + http_client->UrlEscape(project_id_);
+    }
+    else {
+      url += "&$userProject=" + http_client->UrlEscape(project_id_);
+    }
+  }
```

Comment:
I'm wondering if we can make this simpler (and likewise for the other places where you need to check the start of the query string). 

For example:
```
bool firstQueryAdded = false;
if (!api_key_.empty()) {
  url += ...
  firstQueryAdded = true;
}
if(!project_id_.empty()){
  url += (firstQueryAdded ? "&" : "?") + ...;
  firstQueryAdded = true;
}
```
What do you think?

Reply:
Agreed, thank you for the suggestion!